types/wlr_output.c

```diff
@@ -698,6 +698,27 @@ static void output_cursor_update_visible(struct wlr_output_cursor *cursor) {
 	cursor->visible = visible;
 }
 
+static bool output_cursor_attempt_hardware(struct wlr_output_cursor *cursor) {
+	struct wlr_texture *texture = cursor->texture;
+	if (cursor->surface != NULL) {
+		texture = cursor->surface->texture;
+	}
+
+	struct wlr_output_cursor *hwcur = cursor->output->hardware_cursor;
+	if (cursor->output->impl->set_cursor && (hwcur == NULL || hwcur == cursor)) {
+		if (cursor->output->impl->move_cursor && hwcur != cursor) {
+			cursor->output->impl->move_cursor(cursor->output,
```

Comment:
Can you assert cursor->output->impl->set_cursor here

Reply:
Ah, you mean a backend implementing `set_cursor` must also implement `move_cursor`? Seems like a thing that should be in `wlr_output_init` too.

Reply:
aye